lib/commands/performance.js

```diff
@@ -0,0 +1,339 @@
+// in case of network info : [[st, bucketDuration, activeTime, rb, rp, tb, tp, op], [1478088000, 3600, null, null, 32115296, 34291, 2956805, 25705 op=0], [1478091600, 3600, null, null, 2714683, 11821, 1420564, 12650 op=0], [1478095200, 3600, null, null, 10079213, 19962, 2487705, 20015 op=0], [1478098800, 3600, null, null, 4444433, 10227, 1430356, 10493 op=0]]
+// in case of cpu info : [[user, kernel], [0.9, 1.3]]
+//
+commands.getPerformanceData = async function (packageName, dataType, dataReadTimeout) {
+
+  if (dataType === 'batteryinfo') {
+    return this.getBatteryInfo(dataReadTimeout);
+  } else if (dataType === 'cpuinfo') {
+    return this.getCPUInfo(packageName, dataReadTimeout);
+  } else if (dataType === 'memoryinfo') {
+    return this.getMemoryInfo(packageName, dataReadTimeout);
+  } else if (dataType === 'networkinfo') {
+    return this.getNetworkTrafficInfo(dataReadTimeout);
+  } else {
+    throw new Error('No performance data of type \'' + dataType + '\' found.');
+  }
+};
+
+helpers.getCPUInfo = async function (packageName, dataReadTimeout) {
+  let returnValue = [];
+  returnValue[0] = ["user", "kernel"];
+
+  let cmd, data, start, end, user, kernel;
+  let loop = true;
+  let tryCount = 0;
+  cmd = ['dumpsys', 'cpuinfo', '|', 'grep', `'${packageName}'`];
+  
+  //sometimes, the function of 'adb.shell' fails. when I tested this function on the target of 'Galaxy Note5', 
+  //adb.shell(dumpsys cpuinfo) returns cpu datas for other application packages, but I can't find the data for packageName.
+  //It usually fails 30 times and success for the next time,  
+  //Since then, he has continued to succeed.
+  while ( loop ) {
+      
+    if (tryCount === (dataReadTimeout + 1) )
+      throw new Error('No response received after ' + dataReadTimeout);
+    else {
+      ++ tryCount;
+      data = await this.adb.shell(cmd);        
+    }
+
+    if (data !== "" && data !== undefined && data !== null){
+      start = data.indexOf(":");  
+      end   = data.indexOf("%", start+1);
+      if (data.indexOf(":", start+1) > 0 && data.indexOf("faults", start+1) < 0)  
+        start = data.indexOf(":", start+1);
+      user = data.substring(start+1, end).trim();
+      start = data.indexOf("+");
+      end = data.indexOf("%", start+1);
+      kernel = data.substring(start+1, end).trim();
+            
+      if (user !== "" && user !== undefined && user !== "nodex"){
+        returnValue[1] = [user, kernel];
+        return returnValue;
+      }
+    }
+    await sleep(1000);
+  }
+};
+
+helpers.getBatteryInfo = async function (dataReadTimeout) {
+  let returnValue = [];
+  returnValue[0] = ["power"];
+  let cmd, data, start, end, power;
+  let loop = true;
+  let tryCount = 0;
+  cmd = ['dumpsys', 'battery', '|', 'grep', "\'level\'"];
+
+  while ( loop ) {
```

Comment:
is there any way to break the loop statement? 
for example, when I call retryInterval(dataReadTimeout, 1000, fn)
in the loop, the function succed to read capability data at the second try, so there is no need to try to call any more.
how to stop the loop?